### ISSUE TITLE
Training model with addition of new parking slot type

### ISSUE THREAD

**Issue by chokkarapu** [Teoge/DMPR-PS#1]
Hello Teoge,

I want to add on more type of OPEN Slot parking type, following steps done

Labelled new type of Open SLOT Type with index number "2"....
Now "0"---> for T Slot, "1"--->for L Shaped and "2"--->Open SLot.
Prepared dataset as per your code recommendations.
Trained the model, with above dataset for 20EPOCHS and finally model got trained.
Now during inference, observed that the marking_points.shape value,
For L-Shaped and Open Shaped value greater then (>) 0.5
T-Shaped value is very less i.e. less then 0.5
I'm unable to differentiate the shapes between L and Open Shape parking slots.
I kindly request you to provide information, how to differentiate between the shape types from model, do we need to change config parameter during training itself so that we get different shape values for different parking shapes.

I request you to provide some information for above regards.

Thanks,
Anil.

**Comment by Teoge** [comment]
Hello, Anil.

In DMPR-PS, we only consider two types of marking-point shape, L-shaped marking-point and T-shaped marking-point. Thus we treat regression of marking-point shape as a binary classification problem and use sigmoid for final prediction. Results less than 0.5 are considered as T-shaped and results greater than 0.5 are considered as L-shaped.

But, in your case, since you have added another shape type, predicting the shape becomes a classification problem with three categories. The original sigmoid no longer works, and you need use softmax instead. Furthermore, corresponding loss function and data strutures in [data/struct.py](https://github.com/Teoge/DMPR-PS/blob/4b6a1c2218bcc7adc1944eeaa191bfa85b8e0867/data/struct.py) also need to be altered. I suggest you read our paper and the code carefully before you proceed any further.

**Comment by chokkarapu** [comment]
Thank you Teogo for your information, will append my changes and figure out for open slots.

**Comment by chokkarapu** [comment]
Hello Teoge,

By taking comments of adding one more shape into consideration, I could finally succeed in these regards.

Im having following concerns on your network architecture:, could you kindly provide your valuable information:
1) Observed that you network architecture is **very large** i.e. around ~24 CONV layers and also your number of channels in each and every layers gets on increases finally landing upto 1024 number of Chanel before finale layer output of 8 Channels.
2) I even observed that your feature map size is 16x16 feature map size
3) I Even observed the input image size is of 512x512 .

Now my queries w.r.t above concerns are as follows:
1) Instead of this big architecture, can we go with small architecture as MARKING point of parking slot is very small in size, so do you have any idea for making it smaller size of network.
2) I believe the feature map size 16x16 is very very small, instead we can go with 32x32 size then computational time gets decrease?
3) I believe the image size can be resized to 256x256 for examples and do it.

Correct me if my observations or modification are wrong.

Can you please provide any more valuable suggestions on above mentioned queries, so that CNN architecture should be lite weighted.


NOTE: I agree, that If we go with any modifications then corresponding functions/modules has to be modify

**Comment by Teoge** [comment]
Hello, Anil.

Your sugestions on modifying the neural network architechture are all feasible. I think these modifications will lead to a small decrease of detection performance but an improvement of detection speed. However, we have not experimented such a configuration thus I cannot be sure. We would appreciate it if you could test this configuration and give us some feedback.

**Comment by chokkarapu** [comment]
Hello Teoge,

Thanks for your response, definitely we will try with mentioned configuration and I'll let you know my findings .

In the data set, some parking slots are of SLANT Type, did you labelled and used for training the network and then done inference for SLANT Type??

I eagerly look forward for your response.

Thanks,
Anil

**Comment by Teoge** [comment]
Hello Anil.

DMPR-PS is unable to detect slant type parking-slots due to its mechanism. We excluded slant type parking-slots from dataset during training and testing.

**Comment by chokkarapu** [comment]
Thanks for prompt response and also for your information....
